spec/controllers/v0/burial_claims_controller_spec.rb

```diff
@@ -38,48 +34,43 @@ def send_create
       expect(monitor).to receive(:track_create_error).once
       expect(form).not_to receive(:process_attachments!)
 
-      response = send_create
+      response = post(:create, params: { param_name => { form: form.form } })
       expect(response.status).to eq(500)
     end
   end
 
   describe '#show' do
-    it 'returns the submission status when the claim uses central mail' do
-      claim = create(:burial_claim_v2)
-      claim.central_mail_submission.update!(state: 'success')
-      get(:show, params: { id: claim.guid })
-
-      expect(JSON.parse(response.body)['data']['attributes']['state']).to eq('success')
-    end
+    let(:claim) { build(:burial_claim_v2) }
 
-    it 'returns the submission status when the claim uses benefits intake' do
-      claim = create(:burial_claim_v2)
-      claim.form_submissions << create(:form_submission, :pending, form_type: '21P-530V2')
-      get(:show, params: { id: claim.guid })
+    it 'returns a success when the claim is found' do
+      allow(SavedClaim::Burial).to receive(:find_by!).and_return(claim)
+      response = get(:show, params: { id: claim.guid })
 
-      expect(JSON.parse(response.body)['data']['attributes']['state']).to eq('success')
+      expect(response.status).to eq(200)
     end
 
     it 'returns an error if the claim is not found' do
       expect(monitor).to receive(:track_show404).once
 
-      get(:show, params: { id: '12345' })
+      response = get(:show, params: { id: 'non-existant-saved-claim' })
 
-      expect(response).to have_http_status(:not_found)
+      expect(response.status).to eq(404)
     end
 
     it 'logs show errors' do
-      allow(SavedClaim::Burial).to receive(:find_by).and_raise(StandardError, 'mock error')
+      error = StandardError.new('Mock Error')
+      allow(SavedClaim::Burial).to receive(:find_by!).and_raise(error)
+
       expect(monitor).to receive(:track_show_error).once
 
-      get(:show, params: { id: '12345' })
+      response = get(:show, params: { id: 'non-existant-saved-claim' })
 
-      expect(response).to have_http_status(:unprocessable_entity)
+      expect(response.status).to eq(500)
     end
   end
 
   describe '#process_and_upload_to_lighthouse' do
-    let(:claim) { build(:pensions_module_pension_claim) }
+    let(:claim) { build(:burial_claim_v2) }
```

Comment:
I do not see the burial_claim factory being used anywhere. Should burial_claim_v2 replace burial_claim?  Looks like formV2 just needs to be added to the burial_claim factory

Reply:
Not sure what you mean by this - `burial_claim_v2` is what should be used here, and is present in the factory `spec/factories/burial_claim.rb` line 36

Reply:
My bad. I see others using the original